src/ProtobufProtocolSupport.chpl

```diff
@@ -905,6 +941,27 @@ module ProtobufProtocolSupport {
       }
       return returnList;
     }
+   
+    proc messageRepeatedAppend(valList, fieldNumber: int, ch: writingChannel) throws {
+      if valList.isEmpty() then return;
+      for val in valList {
+        tagAppend(fieldNumber, lengthDelimited, ch);
+        messageAppendBase(val, ch);
+      }
+    }
+
+    proc messageRepeatedConsume(ch: readingChannel, type messageType) throws {
+      var returnList: list(messageType);
+      var tmpMem = openmem();
+      var memWriter = tmpMem.writer(kind=iokind.little, locking=false);
+      var memReader = tmpMem.reader(kind=iokind.little, locking=false);
+      
+      var tmpObj: messageType;
+      messageConsumeBase(ch, tmpObj, memWriter, memReader);
+      returnList.append(tmpObj);
+      tmpMem.close();
+      return returnList;
```

Comment:
This doesn't look like it loops, should it?  The other repeatedConsumes seem like they do.  Do you have a test of longer repeated fields?

Reply:
> This doesn't look like it loops, should it?

No, it should not loop. Message types are `lengthDelimited`  types in itself just like `bytes` and `string`. Please see -> https://github.com/Aniket21mathur/Chapel-protobuf/pull/22/files#diff-82681d009c18bffcf9803ad6109f77f8R749. These (bytes, string, and message types)have a different way of appending in repeated fields, the other types need to be wrapped in a `lengthDelimited` type since all repeated fields should be `lengthDelimited`.

For message, bytes and string you can observe that we are appending tag multiple times, a new tag with the same field number for each value from the list, while for other types we have a single tag of the same field number for all the values.

So in this case instead of having a loop this consume function will be called again and again whenever a `messagetype` of the same field number is found, while for other types the consume function(called once)will have to loop over all the values.

These by default length delimited types were the reason, I used `list.extend` in the `readFrom` function for repeated fields.
 

> Do you have a test of longer repeated fields?

Yes, I have tests for longer repeated fields. Not for message types, but for similar bytes and string types.

Reply:
Okay, sounds good :)